source/extensions/filters/network/dubbo_proxy/app_exception.cc

```diff
@@ -20,17 +20,23 @@ AppException::ResponseType AppException::encode(MessageMetadata& metadata,
                                                 DubboProxy::Protocol& protocol,
                                                 Deserializer& deserializer,
                                                 Buffer::Instance& buffer) const {
+  ASSERT(buffer.length() == 0);
+
   ENVOY_LOG(debug, "err {}", what());
 
+  // Serialize the response content to get the serialized response length.
+  const std::string& response = what();
+  size_t serialized_body_size = deserializer.serializeRpcResult(buffer, response, response_type_);
```

Comment:
Need to check whether the response_type_ meets the requirements, it can carry the body

Reply:
esponse_type_ has been assigned ResponseWithException in the constructor,  I'll add an ASSERT.